enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,885 @@
+configured in the cluster. When it isn't possible to use a registry server, the
+administrator of the cluster will explicitly configure the cluster version
+operator to use an upgrade bundle instead, setting the new
+`spec.desiredUpdate.imageSource.sourceType` field to `Bundle` (the default will
+be `Registry`):
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ClusterVersion
+metadata:
+  name: version
+spec:
+  desiredUpdate:
+    imageSource:
+      sourceType: Bundle
+```
+
+The technician that performs the upgrade will receive the USB stick containing
+the upgrade bundle and will plug it in one of the nodes of the cluster.
+
+The cluster version operator will be monitoring device events in all the nodes
+of the cluster in order to detect when such an USB stick (or any other kind of
+media containing an upgrade bundle) has been plugged. To control that a new
+`spec.desiredUpdate.imageSource.bundle.detectionMechanism` field will be added
+to the `ClusterVersion` object. The default value will be `Manual` so that this
+monitoring will be disabled by default. When set to `Automatic` the cluster
+version operator will create a new `bundle-monitor` daemon set that will
+perform the actual monitoring. When any of the pods in this daemon set detects
+a device containing a valid upgrade bundle it will update the status of the
+`ClusterVersion` object indicating that the bundle is available.
+
+For example, if the administrator of the cluster wants to enable automatic
+detection of upgrade bundles she will add this to the `ClusterVersion` object:
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ClusterVersion
+metadata:
+  name: version
+spec:
+  desiredUpdate:
+    imageSource:
+      sourceType: Bundle
+      bundle:
+        detectionMechanism:
+          mechanismType: Automatic
+```
+
+For situations where it isn't desirable to monitor device events, or it isn't
+possible to plug the USB stick or any other kind of media, we will also add a
+new `spec.desiredUpdate.imageSource.bundle.file` field to explicitly indicate
+the location of the bundle. When this is used the value will be directly copied
+to `status.desired.bundle.file` without creating the `bundle-monitor` daemon
+set.
+
+If the administrator wants to disable automatic detection of the upgrade bundle,
+and wants to manually copy the file to one of the cluster nodes she will do
+something like this:
+
+```yaml
+spec:
+  desiredUpdate:
+    imageSource:
+      sourceType: Bundle
+      bundle:
+        detectionMechanism: Manual
+        manual:
+          file: /root/upgrade-4.13.7-x86_64.tar
+```
+
+If the administrator wants to disable automatic detection, but wants to use a
+USB stick anyhow he will do something like this:
+
+```yaml
+spec:
+  desiredUpdate:
+    imageSource:
+      sourceType: bundle
+      bundle:
+        detectionMechanism: Manual
+        manual:
+          file: /dev/sdb
+```
```

Comment:
In the first phase of the feature we should only go with the manual detection mechanism and later we should think about making it automatic

Reply:
Actually we are suggesting a first phase (dev preview -> tech preview) that will only support pre-loading and pinning of the images, without the bundle. The use of the bundle would then be added in the second phase (tech preview -> GA). See the graduation criteria for details. It makes sense to implement automatic detection later. I will add a `After GA` section to the graduation criteria, let me know if that works.